wandb/apis/public/api.py

```diff
@@ -740,11 +759,21 @@ def default_entity(self) -> Optional[str]:
 
     @property
     def viewer(self) -> "public.User":
-        """Returns the viewer object."""
+        """Returns the viewer object.
+
+        Raises:
+            ValueError if viewer data is not able to be fetched from W&B.
+        """
         if self._viewer is None:
-            self._viewer = public.User(
-                self._client, self._client.execute(self.VIEWER_QUERY).get("viewer")
-            )
+            viewer = self._client.execute(self.VIEWER_QUERY).get("viewer")
+
+            if viewer is None:
+                raise ValueError(
+                    "Unable to fetch user data from W&B, "
+                    + "please verify your API key is valid."
```

Comment:
Should use implicit string concatenation instead of `+` (and nit: put the connecting space at the start of the next string)